### ISSUE TITLE
Api should be versionable

### ISSUE THREAD

**Issue by addshore** [JeroenDeDauw/QueryrAPI#1]
This is one of the biggest issues I have with the current MW and WB apis.

**Comment by JeroenDeDauw** [comment]
Yeah, this is an important thing to get right for serious third party usage. The approach taken here makes it a lot easier to do versioning then in the WB API:
- No single "one fits all" format, that needs to be modified whenever something new is added
- REST style content negotiation [versioning such as GHs](https://developer.github.com/v3/media/#api-v3-media-type-and-the-future) is possible
- No technical and consistency restrictions imposed by a wider framework

That said, I do not think implementing versioning is the first thing to start with. Beginning with a explicitly beta/unstable API to create an initial feature set and get feedback is what I have in mind. Then, once people actually care about stability, the switch out of beta-ness can be made, which will involve deciding on a versioning strategy.
